hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/operations/AbstractRingBufferOperation.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.ringbuffer.impl.operations;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.ringbuffer.impl.RingbufferContainer;
+import com.hazelcast.ringbuffer.impl.RingbufferService;
+import com.hazelcast.spi.AbstractOperation;
+
+import java.io.IOException;
+
+import static com.hazelcast.ringbuffer.impl.RingbufferDataSerializerHook.F_ID;
+import static com.hazelcast.ringbuffer.impl.RingbufferService.SERVICE_NAME;
+
+public abstract class AbstractRingBufferOperation extends AbstractOperation
+        implements IdentifiedDataSerializable {
+
+    protected String name;
+    private RingbufferContainer ringbuffer;
+
+    public AbstractRingBufferOperation() {
+    }
+
+    public AbstractRingBufferOperation(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public String getServiceName() {
+        return SERVICE_NAME;
+    }
+
+    public RingbufferContainer getRingBufferContainer() {
```

Comment:
why is it public? I understand it's inside the impl package, but still I wouldn't expose it to a world.